core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -353,6 +353,16 @@ default void createTable(ConnectorSession session, ConnectorTableMetadata tableM
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables");
     }
 
+    /**
+     * Drops the specified table
+     *
+     * @throws RuntimeException if the table cannot be dropped
+     */
+    default void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        dropTable(session, getTableHandle(session, schemaTableName));
```

Comment:
the default implementation should work correctly with redirects
this will allow the caller (the drop task) to just call the new method

add something like

```java
Optional<CatalogSchemaTableName> redirected  = redirectTable(session, schemaTableName);
        if (redirected.isPresent()) {
            return new DropRedirected(schemaTableName);
        }
```

Reply:
handling redirects directly inside dropTable prevents us from invoking access control with proper (final) name before drop is invoked